### ISSUE TITLE
Using PyPlot with latest Julia master infinite loop?

### ISSUE THREAD

**Issue by colbec** [JuliaPy/PyPlot.jl#325]
In:
```
Julia Version 0.7.0-DEV.2272
Commit 0385a60fc7 (2017-10-22 19:38 UTC)
```
`Pkg.update()` followed by `using PyPlot `results in hang at
```
...
WARNING: Base.Range is deprecated, use Base.AbstractRange{T} where T instead.
  likely near /home/colin/.julia/v0.7/PyPlot/src/PyPlot.jl:316
WARNING: Base.Range is deprecated, use Base.AbstractRange{T} where T instead.
  likely near /home/colin/.julia/v0.7/PyPlot/src/PyPlot.jl:316
^CERROR: InterruptException:
Stacktrace:
 [1] process_events at ./libuv.jl:82 [inlined]
 [2] wait() at ./event.jl:262
 [3] wait(::Condition) at ./event.jl:50
 [4] stream_wait(::Base.Process, ::Condition, ::Vararg{Condition,N} where N) at ./stream.jl:42
 [5] wait at ./process.jl:817 [inlined]
 [6] success(::Base.Process) at ./process.jl:686
 [7] compilecache(::String) at ./loading.jl:655
 [8] _require(::Symbol) at ./loading.jl:450
 [9] require(::Symbol) at ./loading.jl:324
```
One CPU goes to 100% and does not budge. Ctrl+c fails to stop the process, and quitting Julia may release the CPU after some time.

**Comment by stevengj** [comment]
Should be fixed by #311.

In general, many packages won't work on 0.7 right now, and if you are on 0.7 you will often need to `Pkg.checkout` the bleeding-edge version.

**Comment by colbec** [comment]
I checked out the latest master of PyPlot and the behaviour is the same.

It looks like it thinks I don't have matplotlib installed and is trying to download a package that never arrives. I have python 2.7 and 3.4 installed, both pip and pip3 report that matplotlib is up to date but matplotlib is only present in site-packages for 3.4 under lib64. PyPlot is using my version 2.7, but wants to get Conda involved. Using ENV to point at Python3.4 has the same result. I have tried adding symbolic links from Python 2.7 site packages both lib and lib64 but behavior is the same. I will try stripping PyPlot from my v0.7 and start again from scratch.

Yes, I am using Julia v0.6 for production where PyPlot is fine, and only report the most bizarre events from v0.7. This one seemed inordinately strange.
